src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs

```diff
@@ -149,6 +150,7 @@ private void LoginButton_OnClick(object sender, RoutedEventArgs e)
 
         private void LogoutOption_Click(object sender, RoutedEventArgs e)
         {
+            if (!DynamoViewModel.IsIDSDKInitialized()) return;
```

Comment:
is there a way to avoid needing to add these checks everywhere? What about adding this check, or an event that eventually raises this check to getters that access data that would have required initialization to succeed in the authprovider - ie the token.

With the current solution proposed here devs need to know that they should add these checks.

Reply:
The first attempt was exactly that, but problem with that is we use those getters at multiple places, so when user clicks on `SignIn` button, the event would be fired 6 times and the popup will be shown 6 times. I met with tibi to discuss a solution, this approach may be more code, but gives more control.